### PR TITLE
Code is covered with unit, widget, integration tests

### PR DESCRIPTION
**Pull request by nikolay-ahramovich-idf**

### CODE REVIEW

File: .github/workflows/build_prod_on_master_commit.yaml [hunk 1]
```diff
@@ -1,10 +1,39 @@
 name: build-apps
 
 on:
-  push:
+  pull_request:
```
Comment (ekaterina-demeshko-idf) [inline .github/workflows/build_prod_on_master_commit.yaml]: correct back to push and GitHub realises

File: integration_test/login_test.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  const splashScreenDelay = Duration(seconds: 3);
+
+  group('Login Test', () {
+    const validLogin = 'test@mail.ru';
+    const validPassword = 'test1234';
+
+    testWidgets('login auth test', (tester) async {
```
Comment (ekaterina-demeshko-idf) [inline integration_test/login_test.dart:14]: is this ok to leave this big anonymous function inside testWidgets?

**Review by ekaterina-demeshko-idf — APPROVED** [review]

File: .github/workflows/build_prod_on_master_commit.yaml (hunk 1 above)
Comment (nikolay-ahramovich-idf) [reply .github/workflows/build_prod_on_master_commit.yaml:4]: I think yes

File: integration_test/login_test.dart (hunk 2 above)
Comment (nikolay-ahramovich-idf) [reply integration_test/login_test.dart:14]: I think yes

File: .github/workflows/build_prod_on_master_commit.yaml (hunk 1 above)
Comment (nikolay-ahramovich-idf) [reply .github/workflows/build_prod_on_master_commit.yaml]: Fixed.

**Review by andrei-lastochkin-idf — APPROVED** [review]

File: packages/data/test/repositories/auth_repository_impl_test.dart [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import 'package:mocktail/mocktail.dart';
+
+class MockAuthService extends Mock implements AuthService {}
+
+void main() {
+  const mockLogin = 'login';
+  const mockPassword = 'password';
+
+  late MockAuthService mockAuthService = MockAuthService();
+  late AuthRepositoryImpl authRepository = AuthRepositoryImpl(
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/test/repositories/auth_repository_impl_test.dart]: 1) why late?
2) why `AuthRepositoryImpl` and not `AuthRepository`?

File: packages/presentation/lib/screens/login/login_bloc.dart [hunk 4]
```diff
@@ -30,6 +30,7 @@ abstract class LoginBloc implements Bloc<BaseArguments, LoginData> {
       );
 
   GlobalKey<FormState> get formStateGlobalKey;
+  set formStateGlobalKey(GlobalKey<FormState> value);
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/screens/login/login_bloc.dart]: define the key in the way u define textcontroller

**Review by sergei-mikhailovskii-idf — APPROVED** [review]
good job

File: packages/presentation/lib/screens/login/login_bloc.dart (hunk 4 above)
Comment (nikolay-ahramovich-idf) [reply packages/presentation/lib/screens/login/login_bloc.dart]: Fixed.

File: packages/data/test/repositories/auth_repository_impl_test.dart (hunk 3 above)
Comment (nikolay-ahramovich-idf) [reply packages/data/test/repositories/auth_repository_impl_test.dart]: 1. I forgot to remove late after some experiments with refactoring.
2. Fixed.

**Pull request merged**